scipy/stats/_stats_py.py

```diff
@@ -84,7 +84,13 @@
            'brunnermunzel', 'alexandergovern']
 
 
+# This should probably be rewritten to avoid nested TypeErrors in favor of
+# branching based on dtype.
 def _contains_nan(a, nan_policy='propagate'):
+    # only inexact (floating/complexfloating) and object arrays support np.nan
+    if isinstance(a, np.ndarray) and not (np.issubdtype(a.dtype, np.inexact) or
```

Comment:
Should we have the first np.ndarray check? When a list which has a nan passed to this function unexpectedly, this returned False without error. This is a private API, so throwing no dtype attribute error might be useful for developers not to forget convert to the ndarray.

Reply:
Good question. I did that because of gh-16318 and the related issue. Let me think about that.

Reply:
What do you think using the build-in `sum`  instead of `np.nan`? It seems that the build-in `sum` works for pd.Series. 
```
In [24]: import numpy as np

In [25]: import pandas as pd

In [26]: np.sum([1,2,np.nan])
Out[26]: nan

In [27]: np.sum(np.array([1,2,np.nan]))
Out[27]: nan

In [28]: np.sum(pd.Series([1,2,np.nan]))
Out[28]: 3.0

In [29]: sum([1,2,np.nan])
Out[29]: nan

In [30]: sum(np.array([1,2,np.nan]))
Out[30]: nan

In [31]: sum(pd.Series([1,2,np.nan]))
Out[31]: nan

In [32]: np.isnan(np.sum(pd.Series([1,2,np.nan])))
Out[32]: False

In [33]: np.isnan(sum(pd.Series([1,2,np.nan])))
Out[33]: True
```

Reply:
Discussing this in gh-16238.